Editing adding_dataset.rst (#1058)
IBM/unitxt: docs/docs/adding_dataset.rst

---------

When we "add a dataset", we are actually adding data for a particiular NLP task such as translation, sentiment classification, question answering, summarization, etc.
In fact, the same dataset can be used for multiple NLP tasks. For example, a dataset with question-and-answer pairs can be used for both
question answering and question generation.  Similarly, a dataset with corresponding English and French sentences can be used for
an Engish-to-French translation task or for a French-to-English translation task.

The task schema is a formal definition of the NLP task.

The `input_fields` of the task are a set of fields that are used to format the textual input to the model.
The `reference_fields` of the task are a set of fields used to format the expected textual output from the model (gold references).
The `metrics` of the task are a set of default metrics used to evaluate the outputs of the model.

While language models generate textual predictions, the metrics often evaluate on different datatypes.  For example,

> Spearman's correlation is evaluated on numeric predictions vs numeric reference, and multi-label F1 is evaluated on a list of string class name prediction_type

Reviewer: ```suggestion
Spearman's correlation is evaluated on numeric predictions vs numeric reference, and multi-label F1 is evaluated on a list of string class name predictions
```

> vs. a reference list of string classes. 

Reviewer: ```suggestion
vs. a list of reference string classes. 
```

> vs. a list of reference string classes. 

Reviewer: ```suggestion
vs. a reference list of string class names.
`

---------

When we "add a dataset", we are actually adding data for a particiular NLP task such as translation, sentiment classification, question answering, summarization, etc.
In fact, the same dataset can be used for multiple NLP tasks. For example, a dataset with question-and-answer pairs can be used for both
question answering and question generation.  Similarly, a dataset with corresponding English and French sentences can be used for
an Engish-to-French translation task or for a French-to-English translation task.

The task schema is a formal definition of the NLP task.

The `input_fields` of the task are a set of fields that are used to format the textual input to the model.
The `reference_fields` of the task are a set of fields used to format the expected textual output from the model (gold references).
The `metrics` of the task are a set of default metrics used to evaluate the outputs of the model.

While language models generate textual predictions, the metrics often evaluate on different datatypes.  For example,

> Spearman's correlation is evaluated on numeric predictions vs numeric reference, and multi-label F1 is evaluated on a list of string class name predictions

Reviewer: ```suggestion
Spearman's correlation is evaluated on numeric predictions vs numeric references, and multi-label F1 is evaluated on a prediction which is a list of string class names
```

    dataset = load_dataset(recipe)

> Or even simpler with Hugging Face datasets:

Reviewer: ```suggestion
Or even simpler with HuggingFace datasets:
```
